cmd/examples/example1.go

```diff
@@ -48,19 +48,32 @@ func main() {
 	}
 
 	// Pass the Filter struct to the FilteredInstanceTypes function of your
-	// selector instance to get a list of filtered instance types and their details
+	// selector instance to get a list of filtered instance types and their details.
 	instanceTypesSlice, err := instanceSelector.FilterInstanceTypes(filters)
 	if err != nil {
 		fmt.Printf("Oh no, there was an error getting instance types: %v", err)
 		return
 	}
 
-	// Pass in your list of instance type details to the appropriate output function
-	// in order to format the instance types as printable strings.
-	maxResults := 100
+	// Pass in the list of instance type details to the SortInstanceTypes function
+	// if you wish to sort the instances based on set filters.
+	sortFilter := "instance-type-name"
+	sortDirection := "ascending"
+	instanceTypesSlice, err = instanceSelector.SortInstanceTypes(instanceTypesSlice, &sortFilter, &sortDirection)
```

Comment:
thinking on this more, I think that we should make this compatible w/ v2 of instance-selector. The original Filter functions could maintain functionality. In fact, I think it makes sense to decouple sorting from the selector pkg all together. It's trivial to sort the `instanceTypesSlice` when using the SDK and is probably a better experience since you know exactly how it's operating.  So sorting should really be part of the CLI which just be a incremental feature that doesn't break backwards compatibility.

Reply:
This does make sense to me how a user of the instance-selector as a library could just manually sort the instance types themselves, so I do agree that sorting could just be part of the CLI. The idea of adding sorting to v2 is also one that I definitely agree with. There is one issue: in order to reduce the number of API calls relating to fetching spot and OD pricing, the pricing caches are only hydrated if the user sets the `PricePerHour` filter. In the current implementation of sorting, if sorting by pricing is desired, pricing caches will be refreshed if needed, but if we were to leave this to the users, then they will not be able to sort by price in call cases. I feel as though this might go against the user expectation of what information is included in the returned instance types. However, this can be solved by mentioning this characteristic of the `PricePerHour` filter in the documentation (comments in filtering method, in `Filters` struct, and possibly the readme). Would just documentation be enough for this or would a different implementation of filtering instance types be necessary?

Reply:
I think docs would be fine for now. If there's a strong signal from users that the limitation is hindering the use of the lib, then we can reevaluate at that time.